RecoEgamma/EgammaPhotonProducers/interface/ReducedEGProducer.h

```diff
@@ -53,6 +53,67 @@ class ReducedEGProducer : public edm::stream::EDProducer<> {
 
+  void linkPhotonCore(const reco::Photon& photon, 
+		      reco::PhotonCoreCollection& photonCores, 
+		      std::map<reco::PhotonCoreRef, unsigned int>& photonCoreMap);
+    
+  void linkSuperCluster(const reco::SuperClusterRef& superCluster, 
+			std::map<reco::SuperClusterRef, unsigned int>& superClusterMap,
+			reco::SuperClusterCollection& superClusters,
+			const bool relink,
+			std::unordered_set<unsigned int>& superClusterFullRelinkMap);
+
+  void linkConversions(const reco::ConversionRefVector& convrefs,
+		       reco::ConversionCollection& conversions, 
+		       std::map<reco::ConversionRef, unsigned int>& conversionMap);
+    
+  void linkConversionsByTrackRef(const edm::Handle<reco::ConversionCollection>& conversionHandle,
+				 const reco::GsfElectron& gsfElectron,
+				 reco::ConversionCollection& conversions, 
+				 std::map<reco::ConversionRef, unsigned int>& conversionMap);
+
+  void linkConversionsByTrackRef(const edm::Handle<reco::ConversionCollection>& conversionHandle,
+				 const reco::SuperCluster& superCluster,
+				 reco::ConversionCollection& conversions, 
+				 std::map<reco::ConversionRef, unsigned int>& conversionMap);
+  
+  void linkConversion(const reco::ConversionRef& convref,
+		      reco::ConversionCollection& conversions, 
+		      std::map<reco::ConversionRef, unsigned int>& conversionMap);
+
+  void linkSeedCluster(const reco::SuperCluster& superCluster, 
+		       reco::CaloClusterCollection & ebeeClusters,
+		       std::map<reco::CaloClusterPtr, unsigned int>& ebeeClusterMap);
+    
+  void linkCaloClusters(const reco::SuperCluster& superCluster, 
+			reco::CaloClusterCollection& ebeeClusters,
+			std::map<reco::CaloClusterPtr, unsigned int>& ebeeClusterMap,
+			std::unordered_set<DetId>& rechitMap,
+			const edm::Handle<EcalRecHitCollection>& barrelHitHandle,
+			const edm::Handle<EcalRecHitCollection>& endcapHitHandle,
+			const CaloTopology *caloTopology,
+			reco::CaloClusterCollection& esClusters,
+			std::map<reco::CaloClusterPtr, unsigned int>& esClusterMap);
+  
+  void relinkCaloCluster(reco::SuperCluster& superCluster, 
+			 const std::map<reco::CaloClusterPtr, unsigned int>& ebeeClusterMap, 
+			 const std::map<reco::CaloClusterPtr, unsigned int>& esClusterMap, 
+			 const edm::OrphanHandle<reco::CaloClusterCollection>& outEBEEClusterHandle, 
+			 const edm::OrphanHandle<reco::CaloClusterCollection>& outESClusterHandle);
+  
+  void relinkSuperCluster(reco::PhotonCore& photonCore, 
+			  const std::map<reco::SuperClusterRef, unsigned int>& superClusterMap, 
+			  const edm::OrphanHandle<reco::SuperClusterCollection>& outSuperClusterHandle);
+
+  void relinkConversions(reco::PhotonCore& photonCore, 
+			 const reco::ConversionRefVector& convrefs,
+			 const std::map<reco::ConversionRef, unsigned int>& conversionMap,
+			 const edm::OrphanHandle<reco::ConversionCollection> &outConversionHandle);
+
+  void relinkPhotonCore(reco::Photon& photon, 
```

Comment:
The order is in which the linking/relinking appears in the code.   I can of course re-organize accordingly.  Should I do the re-ordering in the implementation as well?